contributing/community/review-comments.rst

```diff
@@ -0,0 +1,177 @@
+
+  **Don't:** ``I don't like how you wrote this code`` - there is no clear explanation why you
+  don't like how it's written.
+
+  **Better:** ``I find it hard to read this code as there many nested if statements, can you make it more
+  readable? by encapsulating some of it's details or maybe adding some comments to explain the overall logic.`` -
+  You explain why you find the code hard to read *and* give some suggestions for improvement.
+
+If a piece of code is in fact wrong, explain why:
+
+    * ``This code doesn't comply with Symfony's CS rules. Please see [...] for details``.
+
+    * ``Symfony 3 still uses PHP 5 and doesn't allow the usage scalar type-hints.``.
+
+    * ``I think the code is less readable now`` - careful here, be sure explain why you think
+      the code is less readable, and maybe give some suggestions?
+
+**Examples of valid reasons to reject:**
+
+    * We tried that in the past (link to the relevant PR) but we needed to revert it for XXX reason.
+
+    * That change would introduce too many merge conflicts when merging up Symfony branches.
+      In the past we've always rejected changes like this.
+
+    * I profiled this change and it hurts performance significantly (if you don't profile, it's an opinion, so we can ignore)
+
+    * Code doesn't match Symfony's CS rules (e.g. ``use array()`` instead of ``[]``)
+
+    * We only provide integration with very popular projects (e.g. we integrate Bootstrap but not your own CSS framework
+
+    * This would require adding lots of code and making lots of changes for a feature that doesn't look so important.
+      That could hurt maintaining in the future.
+
+Asking for changes
+------------------
+
+Rarely something is perfect from the start, while the code itself is good.
+It may not be optimal or conform the Symfony coding style.
+
+Again, understand the author already spent time on the issue and asking
+for (small) changes may be misinterpreted or seen as a personal attack.
+
+Be thankful for their work (so far), stay positive and really help them
+to make the contribution a great one. *Especially if they are a first
+time contributor.*
+
+Use words like "Please", "Thank you" and "Could you" instead of making demands;
+
+* "Thank you for your work so far. I left some suggestions for improvement
+  to make the code more readable."
+
+* "Your code contains some coding-style problems, can you fix these before
+  we merge? Thank you"
+
+* "Please use 4 spaces instead of tabs", "This needs be on the previous line";
+
+  During a pull request review you can usually leave more then one comment,
+  you don't have to use "Please" all the time. But it wouldn't hurt.
+
+It may not seem like much, but saying "Thank you" does make others feel
+more welcome.
+
+Using humor
+-----------
+
+In short: Don't be a troll; This violates the Code of Conduct and may
+even get you banned! Keep it real and friendly.
```

Comment:
Do we need to point this out? It sounds a bit aggressive to me, but I'm not sure what others think :)

Reply:
@weaverryan this goes toward my point about cultural context, so yeas if we're trying to create a balance it is probably best left out. Furthermore, I don't think there is a formalised CoC yet … amazing work & progression being made (:heart:) but nothing to back up that assertion with.

Reply:
well we do not yet have a formal CoC for the community (we do at the symfony live/con), but even in the abscene of a formal CoC, we have a factual informal one .. in the sense that even if we don’t adopt a CoC there are limits to what behavior will be permitted. but the term CoC is I guess assumed to mean “formal CoC”, so in that spirit I guess might need to be rephrased to something like `Extreme misbehavior will not be tolerated and may even get you banned`